In [1]:
import pandas as pd
import scipy.stats as stats
import numpy as np

# Question 1

In [3]:
Diet = pd.read_csv('dietstudy.csv')

## Check triglyceride level

In [15]:
print('The triglyceride level before diet plan is : ', format(Diet.tg0.mean()))
print('The triglyceride level after diet plan is : ', format(Diet.tg4.mean()))

The triglyceride level before diet plan is :  138.4375
The triglyceride level after diet plan is :  124.375


## We have 2 samples. Thus, we can use T-Test Two Sample Hypothesis Testing <br>
 -  H0 : triglyceride level before diet == triglyceride level after diet
 -  H1 : triglyceride level before diet < triglyceride level after diet

In [9]:
testTriLvl = stats.ttest_rel(a = Diet.tg0,
                            b = Diet.tg4)

testTriLvl

Ttest_relResult(statistic=1.2000008533342437, pvalue=0.24874946576903698)

In [12]:
if testTriLvl.pvalue > 0.05:
    print('WE FAIL TO REJECT NULL HYPTHESIS')
else:
    print('WE REJECT NULL HYPOTHESIS')
        

WE FAIL TO REJECT NULL HYPTHESIS


## Since H1 is rejected, we accept the H0. Therefore we can say that triglyceride level have stayed the same before and after the diet plan.

## Check Weight before and after diet plan

In [16]:
print('Weight before diet is : ', format(Diet.wgt0.mean()))
print('Weight after diet is : ', format(Diet.wgt4.mean()))

Weight before diet is :  198.375
Weight after diet is :  190.3125


 - H0 : Weight before diet == Weight after diet
 - H1 : Weight before diet <> Weight after diet

In [18]:
testWeight = stats.ttest_rel(a = Diet.wgt0,
                            b = Diet.wgt4)

testWeight

Ttest_relResult(statistic=11.174521688532522, pvalue=1.137689414996614e-08)

In [19]:
if testWeight.pvalue > 0.05:
    print('WE FAIL TO REJECT NULL HYPTHESIS')
else:
    print('WE REJECT NULL HYPOTHESIS')
        

WE REJECT NULL HYPOTHESIS


## Since We reject null hypothesis , We can say that the weight after diet has decreased from 198.37 to 190.31

<br>

<br>

# Question 2.

In [20]:
credit = pd.read_csv('creditpromo.csv')

In [21]:
credit

id         insert      dollars
0       148       Standard  2232.771979
1       572  New Promotion  1403.807542
2       973       Standard  2327.092181
3      1096       Standard  1280.030541
4      1541  New Promotion  1513.563200
..      ...            ...          ...
495  130163  New Promotion  1513.060134
496  130204  New Promotion  1020.757943
497  130255       Standard  1919.856304
498  130583       Standard  1863.014814
499  130682       Standard  1295.012183

[500 rows x 3 columns]

In [26]:
standard_promo = credit.loc[(credit['insert'] =='Standard')]

In [27]:
new_promo = credit.loc[(credit['insert'] == 'New Promotion')]

In [29]:
Avg_standard = standard_promo.dollars.mean()
Avg_new = new_promo.dollars.mean()

In [31]:
print('Average spend by customers getting standard seasonal ad : ', format(Avg_standard))
print('Average spend by customers getting new peomotion ad : ', format(Avg_new))

Average spend by customers getting standard seasonal ad :  1566.3890309659348
Average spend by customers getting new peomotion ad :  1637.4999830647992


## Since we are to test influence of categorical variable (insert) on continous variable (dollar) we would use Independant Sample T-Test <br>

## Two assumptions : -  Amount spend after seasonal ad and amount spend after new promo as are of equal variance 
##                                  - Amount spend in both cases have unequal variance  

If the T-score is same we will accept equal variance. 

 - H0 : Average spend by customers getting both type of ads are equal
 - H1 : Average spend by customers getting both type of ads are not equal

In [33]:
eq_var_test = stats.ttest_ind(a = standard_promo.dollars,
                             b = new_promo.dollars,
                             equal_var= True)

eq_var_test

Ttest_indResult(statistic=-2.2604227264649963, pvalue=0.024225996894147814)

In [34]:
uneq_var_test = stats.ttest_ind(a = standard_promo.dollars,
                             b = new_promo.dollars,
                             equal_var= False)

uneq_var_test

Ttest_indResult(statistic=-2.260422726464996, pvalue=0.024226348191648994)

## Since the t-score in both case are equal, we will consider the equal variance p-value

In [36]:
eq_var_test.pvalue < 0.05


True

## Since p value is low, we reject null hypothesis. Therefore, we can say that the New Promotional ad has been beneficial for the store

<br>

<br>

# Question 3.

In [2]:
polli = pd.read_csv('pollination.csv')

In [3]:
seedmean = polli.Seed_Yield_Plant.mean()
seedmean

180.8035

## a.

 - H0 : overall population of seed yield/plant == 200
 - H1 : overall population of seed yield/plant != 200

## Since we have population mean = 200,  sample size is less than 30 and we have to find whether the sample mean == population mean we will use One Sample T Test

In [4]:
testSeed = stats.ttest_1samp(a = polli.Seed_Yield_Plant, popmean= 200)
testSeed

Ttest_1sampResult(statistic=-2.3009121248548645, pvalue=0.032891040921283025)

In [5]:
testSeed.pvalue > 0.05

False

## Since, p value is low, we reject nul hypothesis. Therefore, overall population of seed yield/plant != 200

## b.

In [6]:
polli

Group  Fruit_Wt  Seed_Yield_Plant  Seedling_length
0   Natural      1.85            147.70            16.86
1   Natural      1.86            136.86            16.77
2   Natural      1.83            149.97            16.35
3   Natural      1.89            172.33            18.26
4   Natural      1.80            144.46            17.90
5   Natural      1.88            138.30            16.95
6   Natural      1.89            150.58            18.15
7   Natural      1.79            140.99            18.86
8   Natural      1.85            140.57            18.39
9   Natural      1.84            138.33            18.58
10     Hand      2.58            224.26            18.18
11     Hand      2.74            197.50            18.07
12     Hand      2.58            230.34            19.07
13     Hand      2.62            217.05            19.00
14     Hand      2.68            233.84            18.00
15     Hand      2.56            216.52            18.49
16     Hand      2.34            211.93            17.45
17     Hand      2.67            210.37            18.97
18     Hand      2.45            199.87            19.31
19     Hand      2.44            214.30            19.36

In [14]:
naturaldf = polli.loc[(polli.Group == 'Natural')]
handdf = polli.loc[(polli.Group == 'Hand')]
print(' Fruit_wt mean of natural pollination is :' , format(naturaldf['Fruit_Wt'].mean()))
print(' Fruit_wt mean of hand pollination is :' , format(handdf['Fruit_Wt'].mean()))

 Fruit_wt mean of natural pollination is : 1.848
 Fruit_wt mean of hand pollination is : 2.5660000000000003


## Comparing Fruit_wt of natural and hand pollination

 - H0 : mean of Fruit_wt is same in natural and hand pollination
 - H1 : Fruit_wt in hand pollination > Fruit in natural pollination
 


In [11]:
fruitWTTest = stats.f_oneway( naturaldf.Fruit_Wt,
                             handdf.Fruit_Wt)

fruitWTTest

F_onewayResult(statistic=312.228532974426, pvalue=8.078362076486568e-13)

In [12]:
if(fruitWTTest.pvalue > 0.05):
    print('We Fail to Reject Null Hypothesis')
else:
    print('We Reject Null Hypothesis')

We Reject Null Hypothesis


## We can say that Fruit_wt has increased when pollination is done by hands.

## Comparing Seed Yield/Plant of hand and natural pollination
 - H0 : mean of Seed Yield/Plant is same in natural and hand pollination
 - H1 : Seed Yield/Plant in hand pollination <> Seed Yield/Plant in natural pollination

In [18]:
print(' Seed Yield/Plant mean of natural pollination is :' , format(naturaldf['Seed_Yield_Plant'].mean()))
print(' Seed Yield/Plant  mean of hand pollination is :' , format(handdf['Seed_Yield_Plant'].mean()))

 Seed Yield/Plant mean of natural pollination is : 146.009
 Seed Yield/Plant  mean of hand pollination is : 215.598


In [16]:
seedYieldTest = stats.f_oneway(naturaldf.Seed_Yield_Plant,
                              handdf.Seed_Yield_Plant)

seedYieldTest

F_onewayResult(statistic=194.83303662980398, pvalue=4.271481585484407e-11)

In [17]:
if(seedYieldTest.pvalue > 0.05):
    print('We fail to reject null hypothesis')
else:
    print('We reject null hypothesis')

We reject null hypothesis


## We reject null hypothesie, therefore we can say that hand pollinated plants have Seed Yield/Plant more than naturally pollinated plants.

## Compare Seedling length
- H0 : mean of Seedling Length is same in natural and hand pollination
- H1 : Seedling Length in hand pollination <> Seedling Length in natural pollination

In [19]:
print('  Seedling length mean of natural pollination is :' , format(naturaldf['Seedling_length'].mean()))
print('  Seedling length  mean of hand pollination is :' , format(handdf['Seedling_length'].mean()))

  Seedling length mean of natural pollination is : 17.706999999999994
  Seedling length  mean of hand pollination is : 18.589999999999996


In [21]:
seedLenghtTest = stats.f_oneway(naturaldf.Seedling_length, handdf.Seedling_length)
seedLenghtTest

F_onewayResult(statistic=6.46293337115627, pvalue=0.020428817064110556)

## Since p < 0.05, we fail to reject null hypothesis, Therefore seedling length is same in natural and hand pollinated plants

<br>

<br>


# Question 4.

In [2]:
dvd = pd.read_csv('dvdplayer.csv')

## Compare the rates given by different age groups 
 - H0 = All age group rated the same
 - H1 = All age group rated diffrently

In [4]:
dvd

agegroup   dvdscore
0   65 and over  38.454803
1         55-64  17.669677
2   65 and over  31.704307
3   65 and over  25.924460
4      Under 25  30.450007
..          ...        ...
63        45-54  46.567682
64  65 and over  23.999491
65     Under 25  24.994419
66  65 and over  33.538502
67        45-54  37.201829

[68 rows x 2 columns]

In [8]:
dvd.agegroup.unique()

array(['65 and over', '55-64', 'Under 25', '25-34', '45-54', '35-44'],
      dtype=object)

In [10]:
AgeGroup1 = dvd['dvdscore'].loc[(dvd['agegroup'] == '65 and over')]
AgeGroup2 = dvd['dvdscore'].loc[(dvd['agegroup'] == '55-64')]
AgeGroup3 = dvd['dvdscore'].loc[(dvd['agegroup'] == '45-54')]
AgeGroup4 = dvd['dvdscore'].loc[(dvd['agegroup'] == '35-44')]
AgeGroup5 = dvd['dvdscore'].loc[(dvd['agegroup'] == '25-34')]
AgeGroup6 = dvd['dvdscore'].loc[(dvd['agegroup'] == 'Under 25')]

In [12]:
AgeTest = stats.f_oneway(AgeGroup1, AgeGroup2, AgeGroup3, AgeGroup4, AgeGroup5, AgeGroup6)
AgeTest

F_onewayResult(statistic=6.992526962676514, pvalue=3.087324905679639e-05)

In [13]:
if AgeTest.pvalue > 0.05:
    print('We fail to reject null hypothesis')
else:
    print('We reject null hypothesis')

We reject null hypothesis


## Since we reject null hypothesis, we can say that the rates given by different age groups is different

<br>


<br>


# Question 6.

In [14]:
sample = pd.read_csv('sample_survey.csv')

In [15]:
sample

id            wrkstat        marital  childs   age  educ  paeduc  \
0        1  Working full time       Divorced     2.0  60.0  12.0    12.0   
1        2  Working part-time  Never married     0.0  27.0  17.0    20.0   
2        3  Working full time        Married     2.0  36.0  12.0    12.0   
3        4  Working full time  Never married     0.0  21.0  13.0     NaN   
4        5  Working full time  Never married     0.0  35.0  16.0     NaN   
...    ...                ...            ...     ...   ...   ...     ...   
2827  2828  Working full time        Married     2.0  73.0  14.0     8.0   
2828  2829  Working full time       Divorced     0.0  40.0  12.0    12.0   
2829  2830            Retired        Married     0.0  73.0   6.0     4.0   
2830  2831  Working full time  Never married     0.0  47.0  12.0    12.0   
2831  2832  Working part-time  Never married     0.0  20.0  10.0     NaN   

      maeduc  speduc          degree  ...        agecat childcat news1 news2  \
0       12.0     NaN     High school  ...      55 to 64      1-2    No    No   
1        NaN     NaN  Junior college  ...      25 to 34     None    No    No   
2       12.0    16.0     High school  ...      35 to 44      1-2    No    No   
3       12.0     NaN     High school  ...  Less than 25     None    No    No   
4       12.0     NaN        Bachelor  ...      35 to 44     None    No    No   
...      ...     ...             ...  ...           ...      ...   ...   ...   
2827    12.0    12.0  Junior college  ...   65 or older      1-2    No    No   
2828    12.0     NaN     High school  ...      35 to 44     None    No    No   
2829     NaN    10.0  LT High school  ...   65 or older     None   Yes    No   
2830    12.0     NaN     High school  ...      45 to 54     None    No    No   
2831     NaN     NaN  LT High school  ...  Less than 25     None    No    No   

      news3 news4 news5      car1      car2      car3  
0        No    No    No  American  Japanese  Japanese  
1       Yes    No    No  American    German  Japanese  
2        No   Yes   Yes  American  American       NaN  
3        No   Yes   Yes  American     Other       NaN  
4        No    No    No  American  American    Korean  
...     ...   ...   ...       ...       ...       ...  
2827     No    No    No  American  American    German  
2828     No    No    No  Japanese       NaN       NaN  
2829    Yes    No    No    Korean  American  American  
2830     No    No    No  Japanese       NaN       NaN  
2831     No    No    No   Swedish  American    German  

[2832 rows x 46 columns]

## a.

## Since we need to find relation between 2 categorical variable. we will use Chi square Test

 - H0 : Observed == Expected , No relation
 - H1 : Observed != Expected , Yes, There is relation

In [18]:
wrmarxtab = pd.crosstab(sample.wrkstat, sample.marital, margins= True)
wrmarxtab

marital                  Divorced  Married  Never married  Separated  Widowed  \
wrkstat                                                                         
Keeping house                  25      200             35         13       55   
Other                          12       16             14          4        8   
Retired                        53      168             17          6      150   
School                          7        9             60          2        1   
Temporarily not working         9       23             11          1        2   
Unemployed, laid off           10       13             32          0        3   
Working full time             295      778            392         58       44   
Working part-time              35      138            102          9       20   
All                           446     1345            663         93      283   

marital                   All  
wrkstat                        
Keeping house             328  
Other                      54  
Retired                   394  
School                     79  
Temporarily not working    46  
Unemployed, laid off       58  
Working full time        1567  
Working part-time         304  
All                      2830

In [20]:
testwrmar = stats.chi2_contingency(observed= wrmarxtab)
testwrmar

(729.2421426572284,
 1.820339965538765e-127,
 40,
 array([[5.16918728e+01, 1.55886926e+02, 7.68424028e+01, 1.07787986e+01,
         3.28000000e+01, 3.28000000e+02],
        [8.51024735e+00, 2.56643110e+01, 1.26508834e+01, 1.77455830e+00,
         5.40000000e+00, 5.40000000e+01],
        [6.20932862e+01, 1.87254417e+02, 9.23045936e+01, 1.29477032e+01,
         3.94000000e+01, 3.94000000e+02],
        [1.24501767e+01, 3.75459364e+01, 1.85077739e+01, 2.59611307e+00,
         7.90000000e+00, 7.90000000e+01],
        [7.24946996e+00, 2.18621908e+01, 1.07766784e+01, 1.51166078e+00,
         4.60000000e+00, 4.60000000e+01],
        [9.14063604e+00, 2.75653710e+01, 1.35879859e+01, 1.90600707e+00,
         5.80000000e+00, 5.80000000e+01],
        [2.46954770e+02, 7.44740283e+02, 3.67109894e+02, 5.14950530e+01,
         1.56700000e+02, 1.56700000e+03],
        [4.79095406e+01, 1.44480565e+02, 7.12197880e+01, 9.99010601e+00,
         3.04000000e+01, 3.04000000e+02],
        [4.46000000e+02, 1.345

In [22]:
print('chi square stats is : ', testwrmar[0])
print('p_value is : ', testwrmar[1])

chi square stats is :  729.2421426572284
p_value is :  1.820339965538765e-127


## Since P-Value is less than 0.05, we reject null hypothesis. Therefore, we can say that there is relation between labour force status and maritial status

## b.
 - H0 : observer == expected, Education qualification does not affect martial status
 - H1 : observed != expected, Education qualification affects martial status

In [24]:
edmarxtab = pd.crosstab(sample.degree, sample.marital)
edmarxtab

marital         Divorced  Married  Never married  Separated  Widowed
degree                                                              
Bachelor              58      251            129         12       28
Graduate              29      123             41          3        9
High school          241      686            367         58      148
Junior college        45      108             46          3        6
LT High school        70      174             77         17       92

In [26]:
edmarTest = stats.chi2_contingency(observed= edmarxtab)
edmarTest

(122.68449020508541,
 1.6707923432360119e-18,
 16,
 array([[ 75.06345268, 227.39312301, 111.83268345,  15.75824176,
          47.95249911],
        [ 32.19248493,  97.52215526,  47.9617157 ,   6.75824176,
          20.56540234],
        [235.55476781, 713.57674583, 350.9393832 ,  49.45054945,
         150.4785537 ],
        [ 32.66359447,  98.94930876,  48.66359447,   6.85714286,
          20.86635945],
        [ 67.52570011, 204.55866714, 100.60262318,  14.17582418,
          43.1371854 ]]))

In [29]:
chiTest = edmarTest[0]
pValue = edmarTest[1]
print('P-Value is : ', pValue)

P-Value is :  1.6707923432360119e-18


In [30]:
pValue > 0.05

False

## Since p-value is less than 0.05, we reject null hypothesis, Therefore we can say that Education qualification affects martials status

## c.
 - H0 : observed == expected , happiness is influenced by earning
 - H1 : observed != expected, happiness is not influenced by earning

In [32]:
hapearxtab = pd.crosstab(sample.happy, sample.income, margins= True)
hapearxtab

income         $1000 TO 2999  $10000 - 14999  $15000 - 19999  $20000 - 24999  \
happy                                                                          
Not too happy              7              39              33              40   
Pretty happy              20             107             119             155   
Very happy                 5              44              26              50   
All                       32             190             178             245   

income         $25000 or more  $3000 TO 3999  $4000 TO 4999  $5000 TO 5999  \
happy                                                                        
Not too happy             113              9              9              6   
Pretty happy              888             11             13             18   
Very happy                571              4             10             11   
All                      1572             24             32             35   

income         $6000 TO 6999  $7000 TO 7999  $8000 TO 9999  LT $1000   All  
happy                                                                       
Not too happy             14             12              9        11   302  
Pretty happy              13             21             30        13  1408  
Very happy                 6             14             19        11   771  
All                       33             47             58        35  2481

In [34]:
hapearTest = stats.chi2_contingency(observed= hapearxtab)
hapearTest

(178.95053061216427,
 7.234749067043371e-21,
 36,
 array([[   3.89520355,   23.12777106,   21.66706973,   29.82265216,
          191.35187424,    2.92140266,    3.89520355,    4.26037888,
            4.01692866,    5.72108021,    7.06005643,    4.26037888,
          302.        ],
        [  18.16041919,  107.82748892,  101.01733172,  139.04070939,
          892.1305925 ,   13.62031439,   18.16041919,   19.86295848,
           18.72793229,   26.67311568,   32.91575977,   19.86295848,
         1408.        ],
        [   9.94437727,   59.04474002,   55.31559855,   76.13663845,
          488.51753325,    7.45828295,    9.94437727,   10.87666264,
           10.25513906,   14.60580411,   18.0241838 ,   10.87666264,
          771.        ],
        [  32.        ,  190.        ,  178.        ,  245.        ,
         1572.        ,   24.        ,   32.        ,   35.        ,
           33.        ,   47.        ,   58.        ,   35.        ,
         2481.        ]]))

In [35]:
hapearTest[1] > 0.05 

False

## Since pvalue is less than 0.05, we reject null hypothesis. Therefore, we can say that happiness is not influenced by earning

- H0 : happiness is influenced by martial status
- H1: happiness is not influenced by martial status

In [37]:
hapmarxtab = pd.crosstab(sample.happy, sample.marital, margins=True)
hapmarxtab

marital        Divorced  Married  Never married  Separated  Widowed   All
happy                                                                    
Not too happy        72       71            108         30       59   340
Pretty happy        278      684            426         49      137  1574
Very happy           93      582            120         13       83   891
All                 443     1337            654         92      279  2805

In [40]:
hapmartest = stats.chi2_contingency(observed=hapmarxtab)
hapmartest

(260.68943894182826,
 7.762777322980048e-47,
 15,
 array([[  53.6969697 ,  162.06060606,   79.27272727,   11.15151515,
           33.81818182,  340.        ],
        [ 248.58538324,  750.24527629,  366.98609626,   51.62495544,
          156.55828877, 1574.        ],
        [ 140.71764706,  424.69411765,  207.74117647,   29.22352941,
           88.62352941,  891.        ],
        [ 443.        , 1337.        ,  654.        ,   92.        ,
          279.        , 2805.        ]]))

In [41]:
hapmartest[1] > 0.05

False

## p-value is less than 0.05, that means, hapiness is also not influenced by martial status

In [42]:
hapearTest[0] > hapmartest[0]

False

## But out of earning and martial status, chi square stats of martial status is more, thus, we can say that martial status affects more than earning